runners/core-java/src/test/java/org/apache/beam/runners/core/InMemoryStateInternalsTest.java

```diff
@@ -157,6 +169,70 @@ public void testMergeBagIntoNewNamespace() throws Exception {
   }
 
   @Test
+  public void testSet() throws Exception {
+    SetState<String> value = underTest.state(NAMESPACE_1, STRING_SET_ADDR);
+
+    // State instances are cached, but depend on the namespace.
+    assertEquals(value, underTest.state(NAMESPACE_1, STRING_SET_ADDR));
```

Comment:
Prefer to use `assertThat` with Hamcrest

```java
assertThat(underTest.state(NAMESPACE_1, STRING_SET_ADDR), equalTo(value));
assertThat(underTest.state(NAMESPACE_2, STRING_SET_ADDR, not(equalTo(value)));
```

and below. (It is not such a big deal for `assertEquals` but the error messages for `assertFalse` and `assertTrue` are not very useful)